src/coreclr/jit/emitarm64.cpp

```diff
@@ -12339,11 +12079,11 @@ BYTE* emitter::emitOutputLoadLabel(BYTE* dst, BYTE* srcAddr, BYTE* dstAddr, inst
         ssize_t imm12 = (ssize_t)dstAddr & 0xFFF; // 12 bits
         assert(isValidUimm<12>(imm12));
         code_t code =
-            emitInsCode(INS_add, IF_DI_2A);  // DI_2A  X0010001shiiiiii iiiiiinnnnnddddd   1100 0000   imm(i12, sh)
-        code |= insEncodeDatasize(EA_8BYTE); // X
-        code |= ((code_t)imm12 << 10);       // iiiiiiiiiiii
-        code |= insEncodeReg_Rd(dstReg);     // ddddd
-        code |= insEncodeReg_Rn(dstReg);     // nnnnn
+            emitInsCode(INS_add, IF_DI_2A);   // DI_2A  X0010001shiiiiii iiiiiinnnnnddddd   1100 0000   imm(i12, sh)
+        code |= insEncodeDatasize(EA_8BYTE);  // X
+        code |= ((code_t)imm12 << 10);        // iiiiiiiiiiii
+        code |= insEncodeReg_R<4, 0>(dstReg); // ddddd
```

Comment:
My only reservation about doing these is that previously `insEncodeReg_Rd` was a little bit safer to use - the caller just needs to know it has an `Rd` field and the callee underneath decides on the bits.
On the other hand, this new version is making the encoding explicit in the caller.

We could add some defines and use those instead?
```
#define insEncodeReg_Rd insEncodeReg_R<4, 0>
#define insEncodeReg_Rn insEncodeReg_V<9, 5>
etc
```

Not sure if that's better or not.

Reply:
I wouldn't mind using the defines. @kunalspathak what do you think?

Reply:
IMO, use inline functions, not `#defines`, if you want helpers.

Reply:
+1 - becomes so much easier while debugging.

Reply:
Got it, I'll revert the deletion of those helper methods, and update them to use the templated implementations.